edn/src/namespaceable_name.rs

```diff
@@ -83,10 +83,38 @@ impl NamespaceableName {
         }
     }
 
+    fn dwim<N, T>(namespace: Option<N>, name: T) -> Self where N: AsRef<str>, T: AsRef<str> {
```

Comment:
This is a pretty poor name; can we do better?  Maybe... `new`?  Also, should this be `Into<Option<AsRef<str>>>`?